app/admin/device-types/edit-result.php

```diff
@@ -36,8 +36,12 @@
 				"tdescription"=>@$_POST['tdescription']);
 
 # update
-if(!$Admin->object_modify("deviceTypes", $_POST['action'], "tid", $values)) 	{ $Result->show("danger",  _("Failed to $_POST[action] device type").'!', false); }
-else 																			{ $Result->show("success", _("Device type $_POST[action] successful").'!', false); }
+if(!$Admin->object_modify("deviceTypes", $_POST['action'], "tid", $values)) {
+    $Result->show("danger", _("Failed to")." ".$_POST[action]." "._("device type").'!', false);
```

Comment:
On line 40. action should be quoted;   `$_POST["action"] `or `"bla $_POST[action] bla"`

`PHP Warning:  Use of undefined constant action - assumed 'action' (this will throw an Error in a future version of PHP)`

Also $_POST('action') will be "add", "edit", "edit-multiple" or "delete"
You may want to translate this in the $Result->show error messages
```php
$Result->show("danger", _("Failed to")." "._($_POST["action"])." "._("device type").'!', false)
```

Reply:
Ahh, thanks for the pointer, and the solution far a question I had in mind about this. I applied your suggested solution.
I've seen already that the translation in such cases wasn't working as wanted and required as the string in `$_POST["action"]` isn't translated. I was thinking about trying to solve this later after reworking on the mass of currently untranslated strings, currently I don't know enough how the l10n stuff is working within PHP in detail.

I guess that quite all of the similar code needs the same preparation like this above?
```
$ git grep -n "\.\$_POST\[action\]"
app/admin/groups/edit-group-result.php:78:if(!$Admin->object_modify("userGroups", $_POST['action'], "g_id", $values)) { $Result->show("danger",  _("Group")." ".$_POST[action]." "._("error")."!", false); }
app/admin/groups/edit-group-result.php:79:else                                                                                                                                          { $Result->show("success", _("Group")." ".$_POST[action]." "._("success")."!", false); }
app/admin/locations/edit-result.php:107:    $Result->show("danger", _("Location")." ".$_POST[action]." "._("failed"), false);
app/admin/locations/edit-result.php:110:    $Result->show("success", _("Location")." ".$_POST[action]." "._("successful"), false);
app/admin/nameservers/edit-result.php:77:if(!$Admin->object_modify("nameservers", $_POST['action'], "id", $values))     { $Result->show("danger", _("Failed to")." ".$_POST[action]." "._("nameserver set").'!', true); }
app/admin/nameservers/edit-result.php:78:else { $Result->show("success", _("Nameserver set")." ".$_POST[action]." "._("successful").'!', false); }
app/admin/powerDNS/domain-edit-result.php:107:    $Result->show("danger", _("Failed to")." ".$_POST[action]." "._("domain").'!', true);
app/admin/powerDNS/domain-edit-result.php:110:    $Result->show("success", _("Domain")." ".$_POST[action]." "._("successful").'!', false);
app/admin/routing/edit-bgp-mapping-submit.php:28:    $Result->show("danger", _("Mapping")." ".$_POST[action]." "._("failed"), false);
app/admin/routing/edit-bgp-mapping-submit.php:31:    $Result->show("success", _("Mapping")." ".$_POST[action]." "._("successful"), false);
app/admin/routing/edit-bgp-submit.php:77:    $Result->show("danger", _("BGP")." ".$_POST[action]." "._("failed"), false);
app/admin/routing/edit-bgp-submit.php:80:    $Result->show("success", _("BGP")." ".$_POST[action]." "._("successful"), false);
app/admin/sections/edit-result.php:113: if(!$Sections->modify_section ($_POST['action'], $values, @$_POST['id']))       { $Result->show("danger", _("Section")." ".$_POST[action]." "._("failed"), false); }
app/admin/sections/edit-result.php:114: else { $Result->show("success", _("Section")." ".$_POST[action]." "._("successful"), false); }
app/admin/subnets/edit-folder-result.php:221:           else { $Result->show("success", _("Folder")." ".$_POST[action]." "._("successful").'!', true); }
app/admin/subnets/edit-result.php:408:          else { $Result->show("success", _("Subnet")." ".$_POST[action]." "._("successful").'!', false); }
app/admin/tags/edit-result.php:54:if(!$Admin->object_modify("ipTags", $_POST['action'], "id", $values))         { $Result->show("danger", _("Tag")." ".$_POST[action]._(" error"), true); }
app/admin/tags/edit-result.php:55:else                                                                                                                                  { $Result->show("success", _("Tag")." ".$_POST[action]._(" success"), false); }
app/admin/users/edit-result.php:166:    $Result->show("danger", _("User")." ".$_POST[action]." "._("failed").'!', true);
app/admin/users/edit-result.php:169:    $Result->show("success", _("User")." ".$_POST[action]." "._("successful").'!', false);
app/admin/vlans/edit-domain-result.php:70:else { $Result->show("success", _("Domain")." ".$_POST[action]." "._("successful").'!', false); }
app/admin/vrf/edit-result.php:82:    $Result->show("danger", _("Failed to")." ".$_POST[action]." "._("VRF").'!', true);
app/admin/vrf/edit-result.php:85:    $Result->show("success", _("VRF")." ".$_POST[action]." "._("successful").'!', false);
app/tools/pstn-prefixes/edit-number-result.php:103:    $Result->show("danger", _("Number")." ".$_POST[action]." "._("failed"), false);
app/tools/pstn-prefixes/edit-number-result.php:106:    $Result->show("success", _("Number")." ".$_POST[action]." "._("successful"), false);
app/tools/pstn-prefixes/edit-result.php:157:    $Result->show("danger", _("Prefix")." ".$_POST[action]." "._("failed"), false);
app/tools/pstn-prefixes/edit-result.php:160:    $Result->show("success", _("Prefix")." ".$_POST[action]." "._("successful"), false);
```
I can address all of these in a later MR because the current MR is already quite a big one.

BTW: How can I do such a syntax check like you've done above?

Reply:
Anything with a php language server plugin will give you php syntax checking.

I'm currently using Visual Studio Code with the PHP Extension Pack, but there are dozens of IDEs with similar capabilities, even VIM. It's a personal preference. (Atom is completely broken for php development, syntax highlighting and xdebug plugins)

Reply:
I've changed my mind on the below. It won't translate the English sentence word by word as the grammar structure will be different.
``` php
$Result->show("danger", _("Failed to")." "._($_POST["action"])." "._("device type").'!', false)
```

Leaning towards adding a new $Result->show function to handle modify object success/error messages.

So the common pattern below...
``` php
# execute update
if(!$Sections->modify_section ($_POST['action'], $values, @$_POST['id'])) { $Result->show("danger",  _("Section $_POST[action] failed"), false); }
else	{ $Result->show("success", _("Section $_POST[action] successful"), false); }
```
becomes ...
```php
# execute update
$ok = $Sections->modify_section ($_POST['action'], $values, @$_POST['id']);
$Result->show_modify_result(_("Sections"), $_POST["action"], $ok, false);

...

public function show_modify_result($obj_name, $action, $action_ok, $die=false) {
    // Output i10n success/failure messages for database modify action
}


```

Reply:
Hmm, is there anything left on my side to do about this?
Should I revert the previous small modification for uniformity on the old style?